src/node/heart.ts

```diff
@@ -55,3 +45,20 @@ export class Heart {
     }
   }
 }
+
+/**
+ * Helper function for the heartbeatTimer.
+ *
+ * If heartbeat is active, call beat. Otherwise do nothing.
+ *
+ * Extracted to make it easier to test.
+ */
+export async function heartbeatTimer(isActive: Heart["isActive"], beat: Heart["beat"]) {
```

Comment:
`isActive` is a private readonly property so that's why I had to pass in individual functions instead of `heart`.

Reply:
I do think we should long-term consider testing this holistically instead since what we really want to know is if the heart works as a whole but we can look into it later.